### ISSUE TITLE
Home Assistant addon

### ISSUE THREAD

**Issue by alekslyse** [jakowenko/double-take#38]
Would it be possible to make a home assistant addon? Since its based on docker it should be an easy task?

**Comment by alekslyse** [comment]
If you look at frigate, they did both; The docker running on home assistant and as an integration with sensors, cameras etc. If I was you start simple. Just support addon first, then add integration afterwards. Supporting addon should be very easy to accomplish.

**Comment by jakowenko** [comment]
I agree, add-on seems easier at first. Integrations look like they need to be written in Python.

**Comment by alekslyse** [comment]
Its not that hard. I have written multiple integrations and its quite
simple if its just passing in sensors etc

**Comment by jakowenko** [comment]
I've added partial HA integration via their [REST API](https://developers.home-assistant.io/docs/api/rest/).

If you're using the beta build you can now pass a Home Assistant config object.

```yaml
home_assistant:
  url: http://192.168.1.1:8123
  token: xxx.xxx-xxx
```

It doesn't look like this method officially registers the entity within HA, since it disappears via a restart, but the next time a detection is found it is recreated. So for the time being, I'm using this and dropping the Node-Red -> HA flow I had before.

Let me know what you think of this for the time being.

<img width="585" alt="Screen Shot 2021-05-17 at 2 14 55 AM" src="https://user-images.githubusercontent.com/1081811/118440870-50c01200-b6b6-11eb-89e5-858b8cb3f880.png">
<img width="398" alt="Screen Shot 2021-05-17 at 2 15 09 AM" src="https://user-images.githubusercontent.com/1081811/118440872-5158a880-b6b6-11eb-9b29-36695d2eed6d.png">

**Comment by Raphael909** [comment]
Hello @jakowenko , What are your thoughts on doing HA integation via  HA's MQTT Discovery. If retain flag is set, then, all of entities gets updated info automatically. 
here : https://www.home-assistant.io/docs/mqtt/discovery/

I use xiaomi-Dafang hacks and they have integrated with HA via Mqtt discover. Might help https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/blob/master/integration/homeassistant/homeassistant.md

**Comment by jakowenko** [comment]
> Hello @jakowenko , What are your thoughts on doing HA integation via  HA's MQTT Discovery. If retain flag is set, then, all of entities gets updated info automatically. 
> 
> here : https://www.home-assistant.io/docs/mqtt/discovery/
> 
> 
> 
> I use xiaomi-Dafang hacks and they have integrated with HA via Mqtt discover. Might help https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/blob/master/integration/homeassistant/homeassistant.md

@Raphael909 love this idea, I'll look into it more. Didn't realize this was possible. Will this add any benefits to what the current HA implementation does? I'm using the HA API at the moment.

**Comment by Raphael909** [comment]
> > Hello @jakowenko , What are your thoughts on doing HA integation via  HA's MQTT Discovery. If retain flag is set, then, all of entities gets updated info automatically.
> > here : https://www.home-assistant.io/docs/mqtt/discovery/
> > I use xiaomi-Dafang hacks and they have integrated with HA via Mqtt discover. Might help https://github.com/EliasKotlyar/Xiaomi-Dafang-Hacks/blob/master/integration/homeassistant/homeassistant.md
> 
> @Raphael909 love this idea, thank you. I'll look into it more, I didn't realize this was possible. Will this add any benefits to what the current HA implementation does? I'm using the HA API at the moment.

I think benefits are two fold.

From your last comment and I quote "It doesn't look like this method officially registers the entity within HA, since it disappears via a restart, but the next time a detection is found it is recreated". At least it might solve this part. Entities will have info and not have to show unavailable upon restart until next info is pushed. (push vs pull model)
Information that your code publishes can be used by other systems and not just by HA. Say Node-red flow for lack of better use case or example.

**Comment by jakowenko** [comment]
Awesome, thanks for the info. All results per camera and name are currently published to MQTT topics. I need to look into it more to see if anything else needs to change to support HA or if I can use what's current being published.

Having the entities retained after restarts would be great.

**Comment by jakowenko** [comment]
@Raphael909, I made the changes we talked about. This is a lot more flexible and now with the `retain` flag, the state is saved upon HA restarts. I dropped support for the HA API since this method adds more value.

This is what my config looks like in HA now. 

```yaml
sensor:
  - platform: mqtt
    name: David
    icon: mdi:account
    state_topic: "double-take/matches/david"
    json_attributes_topic: "double-take/matches/david"
    value_template: "{{ value_json.camera }}"
```

**Comment by Haloooch** [comment]
I don't fully understand this but if I'm reading the following doc correct if you set the topic to be 'homeassistant/sensor/david' AND HA has MQTT discovery enabled, which it is by default, then there would be no need to manually add the sensor, HA would do it automatically. I think....
https://www.home-assistant.io/docs/mqtt/discovery/

**Comment by jakowenko** [comment]
@Haloooch, thanks for the tip on this. I wasn't aware they could be auto discovered. Looks like a /config and /state topic are needed. I have it working locally. Do you think this should automatically be published when MQTT is enabled? Or should this require an additional config flag for the user to enable?

**Comment by Haloooch** [comment]
Personally, I would be happy with automatic publishing. However, if it can be done through a config flag it would make more sense and keep all users happy. Options always seem to go down better than forcing a solution :)

**Comment by jakowenko** [comment]
I'm good with automatic as well, maybe I'll enable it by default if MQTT is on and allow the user to turn it off if they'd like.

Do you think cameras should be published as well? i.e. `homeassistant/sensor/back-door` with the value of the last person that was found?

**Comment by Haloooch** [comment]
Hi, think cameras should be published. If a user doesn't want them they can always be disabled directly through HA.

**Comment by jakowenko** [comment]
Hey @Haloooch, I pushed this feature to the beta Docker tag if you want to give it a try and let me know what you think. Names and cameras are published under `homeassistant/sensor/${name || camera }/state`.

**Comment by Haloooch** [comment]
Partial success. Got the beta version up and running without issue, I can see the new topics are being published both through MQTT explorer and directly within the HA logs:

```
2021-07-10` 09:41:32 DEBUG (MainThread) [homeassistant.components.mqtt] Received message on homeassistant/sensor/front_door/config (retained): b'{"name":"front_door","icon":"mdi:camera","value_template":"{{ value_json.personCount }}","state_topic":"homeassistant/sensor/front_door/state","json_attributes_topic":"homeassistant/sensor/front_door/state","availability_topic":"double-take/available"}'

2021-07-10 09:41:32 DEBUG (MainThread) [homeassistant.components.mqtt.discovery] Process discovery payload {'name': 'front_door', 'icon': 'mdi:camera', 'value_template': '{{ value_json.personCount }}', 'state_topic': 'homeassistant/sensor/front_door/state', 'json_attributes_topic': 'homeassistant/sensor/front_door/state', 'availability_topic': 'double-take/available', 'platform': 'mqtt'}
2021-07-10 09:41:32 INFO (MainThread) [homeassistant.components.mqtt.discovery] Found new component: sensor front_door

2021-07-10 09:41:43 DEBUG (MainThread) [homeassistant.components.mqtt] Subscribing to homeassistant/sensor/front_door/state, mid: 14
2021-07-10 09:41:43 DEBUG (MainThread) [homeassistant.components.mqtt] Received message on homeassistant/sensor/front_door/state (retained): b'{"id":"1625824871.29101-k0zmn1","duration":54.95,"timestamp":"7/9/2021, 8:02:35 PM","attempts":12,"camera":"front_door","zones":[],"matches":[],"unknown":{"name":"unknown","confidence":0,"match":false,"box":{"top":41,"left":221,"width":47,"height":55},"type":"snapshot","duration":5.08,"detector":"deepstack","filename":"418ab794-60ec-499d-a183-caaa0c283747.jpg"},"personCount":0}'

2021-07-10 09:41:44 DEBUG (MainThread) [homeassistant.components.mqtt] Subscribing to homeassistant/sensor/front_door/state, mid: 28
2021-07-10 09:41:44 DEBUG (MainThread) [homeassistant.components.mqtt] Received message on homeassistant/sensor/front_door/state (retained): b'{"id":"1625824871.29101-k0zmn1","duration":54.95,"timestamp":"7/9/2021, 8:02:35 `PM","attempts":12,"camera":"front_door","zones":[],"matches":[],"unknown":{"name":"unknown","confidence":0,"match":false,"box":{"top":41,"left":221,"width":47,"height":55},"type":"snapshot","duration":5.08,"detector":"deepstack","filename":"418ab794-60ec-499d-a183-caaa0c283747.jpg"},"personCount":0}'
```

However, no device is being automatically created in HA. Still playing to make sure it's not a problem natively in HA.

**Comment by jakowenko** [comment]
Strange, that hasn't been the case for me. When I go to the developer page in HA and search for the sensors they show up. I'm just using the stock MQTT integration in HA. Did you configure yours differently?

**Comment by Haloooch** [comment]
Going through the developer page I can see all sensors are there! What was confusing me was I was expecting to see then listed under Integrations>Mosquitto broker. I can see my other MQTT devices there but wondering if discovered devices don't go there automatically.

One thing I noticed when looking at the settings of the sensor, there is a message:

```
This entity ("sensor.front_bins") does not have a unique ID, therefore its settings cannot be managed from the UI.
```

While it does have an ID ```1625924881.155285-mz9dab```, it's not unique, the sensor that was created for unknown person 'sensor.unknown' has the same id. I'm wondering however if that is expected and not an issue, at least I think that's what this page is saying: https://www.home-assistant.io/faq/unique_id 

Regardless of all the above. I can confirm the sensors are being auto created! Thank you :)

**Comment by jakowenko** [comment]
@Haloooch, thanks for catching that. I verified that passing in a `unique_id` to the `/config` topic fixes that issue. I'll get this addressed in the next release which should be in a few days.

I was thinking of prefixing the `unique_id` with `double_take` or something like that to prevent collisions with other HA entities. Do you have any preferences? Ultimately, it doesn't probably matter as long as it is unique and stays consistent for the matched names and camera names.

While on that topic, are you ok with the sensors just being `sensor.name` and `sensor.camera` or should I make them a little more specific to this application?

**Comment by Haloooch** [comment]
The unique ID is rarely seen in the front end (I think), as such having it tagged with ```double_take``` wouldn't add much value to me as a user, but it might make life easier for troubleshooting at some point. As such if it's easy to do you might as well do it.

Sensor name is an interesting one, I now have a sensor called ```sensor.unknown``` from the camera detecting an unknown person. It just doesn't look right. For this one having ```sensor.double_take_uknown``` would make a lot of sense, at the end of the day the user can always change it to something else if they don't like it but at least its easily recognised when first created.

**Comment by jakowenko** [comment]
@Haloooch, the new beta build should include prefixing the `entity_id` and now includes a `unique_id` too. Both start with `double_take_`. I did notice that the HA `entity_id` comes from the name of the sensor and not the topic name like I originally though, which means the default names of the entities in HA include the prefix. But this can be changed through the UI or YML. I tried passing a `friendly_name` in the topic message, but that didn't seem to do anything.

**Comment by jakowenko** [comment]
I think the MQTT auto discovery should cover most use cases for the HA addon. Closing as this is now included in release [v0.9.0](https://github.com/jakowenko/double-take/releases/tag/v0.9.0). Feel free to open a new issue if something isn't working as intended.
